src/firefly/js/util/VOAnalyzer.js

```diff
@@ -0,0 +1,337 @@
+
+                if (idxs) {
+                    this.setCenterColumnsInfo(cols[0].name, cols[1].name, idxs.lonIdx, idxs.latIdx,
+                                             posCol[UCDCoord.eq.key].coord);
+                }
+            }
+        }
+
+        return this.centerColumnsInfo;
+    }
+
+    /**
+     * get center columns pair by checking the table meta
+     * @returns {null|{lonCol: *, latCol: *, lonIdx: *, latIdx: *, csys: *}|*}
+     */
+    getCenterColumnsOnMeta() {
+        if (this.centerColumnsInfo) {
+            return this.centerColumnsInfo;
+        }
+
+        const {tableMeta} = this.tableModel || {};
+
+        if (!tableMeta ||
+            (!tableMeta[MetaConst.CATALOG_COORD_COLS] && !tableMeta[MetaConst.CENTER_COLUMN])) {
+            return this.centerColumnsInfo;
+        }
+
+        const cenData= tableMeta[MetaConst.CATALOG_COORD_COLS] || tableMeta[MetaConst.CENTER_COLUMN];
+
+        let s;
+        if (cenData) s= cenData.split(';');
+        if (!s || s.length !== 3) {
+            return this.centerColumnsInfo;
+        }
+
+
+        const idxs = getLonLatIdx(this.tableModel, s[0], s[1]);
+        if (idxs) {
+            this.setCenterColumnsInfo(s[0], s[1], idxs.lonIdx, idxs.latIdx, CoordinateSys.parse(s[2]));
+        }
+
+        return this.centerColumnsInfo;
+    }
+
+    /**
+     * get center columns pair by guessing
+     * @returns {null|{lonCol: *, latCol: *, lonIdx: *, latIdx: *, csys: *}|*}
+     */
+    guessCenterColumns() {
+        if (this.centerColumnsInfo) {
+            return this.centerColumnsInfo;
+        }
+        const guess = (lon, lat) => {
+            const lonCol = getColumn(this.tableModel, lon);
+            const latCol = getColumn(this.tableModel, lat);
+            if (lonCol && latCol) {
+                const idxs = getLonLatIdx(this.tableModel, lonCol.name, latCol.name);
+                if (idxs) {
+                    this.setCenterColumnsInfo(lonCol.name, latCol.name, idxs.lonIdx, idxs.latIdx);
+                }
+            }
+            return this.centerColumnsInfo;
+        };
+        guess('ra','dec') || guess('lon', 'lat') || guess('s_ra', 's_dec');
+
+        return this.centerColumnsInfo;
+    }
+
+    /**
+     * return center position or catalog coordinate columns and the associate*d coordinate system
+     * by checking table meta, UCD values or ObsCore columns, otherwise by guessing if none of previous
+     * methods works.
+     * @returns {{lonCol, latCol, csys}|*}
+     */
+    getCenterColumns() {
+
+        return  this.getCenterColumnsInfo() ||
+                this.getCenterColumnsOnMeta() ||
+                this.getCenterColumnsOnUCD() ||
+                this.getCenterColumnsOnObsCore() ||
+                this.guessCenterColumns();
+    }
+
+
+    static newInstance(tableModel) {
+        return new TableRecognizer(tableModel);
+    }
+}
+
+const tableRecognizerMap = new Map();
```

Comment:
The computation is light though the  function for getting center columns is repeatedly called while rendering the table and coverage. Since we didn't come into a firm conclusion in the discussion with @robyww & @loitly  about this, so I just kept the map. The key issue is to decide if we want to keep  the table center columns info or not, the implementation is doable if the info is to be kept.